### ISSUE TITLE
[no-var-requires] require created by createRequire should be ignored

### ISSUE THREAD

**Issue by dnalborczyk** [typescript-eslint/typescript-eslint#4035]
similar to: https://github.com/typescript-eslint/typescript-eslint/issues/3713

```JSON
{
  "rules": {
    "@typescript-eslint/no-var-requires": 2
  }
}
```

```TS
import { createRequire } from 'module'

const require = createRequire(import.meta.url)

const json = require('./some.json')
```

**Expected Result**

**Actual Result**

```
error  Require statement not part of import statement  @typescript-eslint/no-var-requires
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.0.1` |
| `node`                             |`17.0.1` |
